tests/vmi_networking_test.go

```diff
@@ -789,6 +790,104 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			table.Entry("without a specific port number", []v1.Port{}, defaultCIDR),
 			table.Entry("with custom CIDR", []v1.Port{}, customCIDR),
 		)
+
+		When("performing migration", func() {
+			var vmi *v1.VirtualMachineInstance
+			var virtHandlerPod *k8sv1.Pod
+
+			ping := func(ipAddr string) error {
+				return tests.PingFromVMConsole(vmi, ipAddr, "-c 1", "-w 2")
+			}
+
+			getVirtHandlerPod := func() (*k8sv1.Pod, error) {
+				node := vmi.Status.NodeName
+				pod, err := kubecli.NewVirtHandlerClient(virtClient).Namespace(flags.KubeVirtInstallNamespace).ForNode(node).Pod()
+				if err != nil {
+					return nil, fmt.Errorf("failed to get virt-handler pod on node %s: %v", node, err)
+				}
+				return pod, nil
+			}
+
+			runMigrationAndExpectCompletion := func(migration *v1.VirtualMachineInstanceMigration, timeout int) {
+				Eventually(func() error {
+					migration, err = virtClient.VirtualMachineInstanceMigration(migration.Namespace).Create(migration)
+					return err
+				}, timeout, 1*time.Second).Should(Succeed())
+
+				Eventually(func() error {
+					migration, err := virtClient.VirtualMachineInstanceMigration(migration.Namespace).Get(migration.Name, &v13.GetOptions{})
+					if err != nil {
+						return err
+					}
+
+					Expect(migration.Status.Phase).ToNot(Equal(v1.MigrationFailed))
+
+					if migration.Status.Phase == v1.MigrationSucceeded {
+						return nil
+					}
+					return fmt.Errorf("Migration is in phase %s", migration.Status.Phase)
+
+				}, timeout, 1*time.Second).Should(Succeed(), fmt.Sprintf("migration should succeed after %d s", timeout))
+			}
+
+			BeforeEach(func() {
+				var err error
+
+				By("Create VMI")
+				vmi = masqueradeVMI([]v1.Port{}, "")
+				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
+
+				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &v13.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				virtHandlerPod, err = getVirtHandlerPod()
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				if vmi != nil {
+					By("Delete VMI")
+					err := virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &v13.DeleteOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					Eventually(func() error {
+						_, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &v13.GetOptions{})
+						return err
+					}, 60*time.Second, 1*time.Second).Should(
+						SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())),
+						"The VMI should be gone within the given timeout",
+					)
+				}
+			})
+			It("preserves connectivity", func() {
```

Comment:
I agree there is no sync between what it says and what it does, but the original intention is still correct and what needs adjusting is the implementation. I will move the migration outside the test itself as we are not testing the migration itself (that should be done on a dedicated migration test) but the connectivity (that it is preserved).

Reply:
I would also prefer to change the context to "It preserves connectivity before&after migration" & "when using post-copy migration" as I had a hard time to understand what is the goal of this test.
I am not sure that using `BeforeEach` is a good choice even though you don't test migration... It should be used for setup and then you should perform the test(check -> migration -> check).

WDYT?

Reply:
We need to test that connectivity is preserved/still-exists after migration. That is the test.
The migration itself if a per-requisite of the test and if it fails, it is a pre-requisite failure and not what we are focusing in testing. The Before* and After* are fixtures, which setup (and teardown) the tests, from my perspective the migration operation is a setup (i.e. given a migrated VMI we can do the test).

The context is not supposed to say what it does, only in what env it is.
"Given a masquerade binding, when migration is performed, the connectivity is preserved."
I think the sentence is clear, more details are in the test itself is someone wants to dig into details.

Obviously the context can always be improved & changed per the commonality with other tests.

Reply:
I somehow thought it test before&after...

Reply:
You mean the connectivity check? It is checked before the migration to assure the connectivity is there, otherwise there is no point in continuing. We could assume it is fine, but the cost of checking is not high and may save us from doing more work. Consider this check like the existence of the VMI in a running state/phase.